src/virtual_elements.ts

```diff
@@ -40,6 +40,49 @@ const ATTRIBUTES_OFFSET = 3;
 const prevAttrsMap = createMap();
 
 
+/**
+ * Removes an attribute name/value pair.
+ * @param attrsArr The array of attributes name/value pairs.
+ * @param name The name of the attribute to remove.
+ * @return The previous value, if any.
+ */
+// tslint:disable-next-line:no-any
+function removeAttr(attrsArr: any[], name: string): boolean {
+  for (let i = attrsArr.length - 2; i >= 0; i -= 2) {
+    if (attrsArr[i] === name) {
+      attrsArr.splice(i, 2);
+      return true;
+    }
+  }
+  
+  return false;
+}
+
+
+/**
+ * Applies the statics. When importing an Element, any existing attributes that
+ * match a static are converted into a static attribute.
+ * @param node The Element to apply statics for.
+ * @param statics The statics array,
+ * @param attrsArr The imported attributes array.
+ */
+function applyStatics(node: HTMLElement, statics: Statics, attrsArr: any[]) {
+  if (!statics) {
+    return;
+  }
+
+  for (let i = 0; i < statics.length; i += 2) {
+    const name = (statics[i]) as string;
+    const value = statics[i + 1];
+    const isNew = !removeAttr(attrsArr, name);
```

Comment:
Do we want to try a `prevAttrsMap` like algorithm? It would take us down to O(n).

Reply:
Hmm, I'm not sure it is worth the code to implement.

Reply:
It's roughly the same number of lines:

```js
if (!statics) {
  return;
}

for (let i = 0; i < statics.length; i += 2) {
  prevAttrsMap[statics[i]] = i + 1;
}

let j = 0;
for (let i = 0; i < attrsArr.length; i += 2) {
  const name = attrsArr[i];
  if (!prevAttrsMap[name]) {
    attrsArr[j++] = name;
    attrsArr[j++] = attrsArr[i + 1];
    delete prevAttrsMap[name];
  }
}
truncateArray(attrsArr, j);

for (const name in prevAttrsMap) {
  updateAttribute(node, name, statics[prevAttrsMap[name]]);
  delete prevAttrsMap[name];
}
```

But I'd be satisfied with just forward iteration. 😉